network/src/connectivity_manager/mod.rs

```diff
@@ -338,6 +350,32 @@ where
             })
             .collect();
 
+        // Limit the number of dialed connections from a Full Node
+        // This does not limit the number of incoming connections
+        // It enforces that a full node cannot have more outgoing connections than `connection_limit`
+        // including in flight dials.
+        let to_connect_size = if let Some(conn_limit) = self.connection_limit {
+            min(
+                conn_limit - self.connected.len() - self.dial_queue.len(),
+                to_connect.len(),
+            )
+        } else {
+            to_connect.len()
+        };
+
+        send_struct_log!(StructuredLogEntry::new_named("dial_eligible_peers")
+            .field(&logging::NETWORK_CONTEXT, &self.network_context)
+            .data("new_peers", &to_connect));
+
+        // If there's nothing to dial, let's just skip the rest of the function
+        if to_connect_size == 0 {
+            return;
+        }
+
+        let to_connect: Vec<_> = to_connect
+            .choose_multiple(&mut self.rng, to_connect_size)
+            .collect();
```

Comment:
if #4684 lands first, you can remove the conditional `if to_connect_size == 0`, since it will just iterate over an empty vec

Reply:
Yeah, I would argue that this would be cleaner to just benefit from the fact that this produces an iter and so you could either eliminate the collect or move the function inline into line 345/382 below
no need for the early return

Reply:
I've gone ahead and put it in the iter.  I've removed the conditional, and the structured log, as it on longer applies to what is actually being connected to.